dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/vitals/VitalMonitor.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.vitals
+
+import kotlin.math.max
+import kotlin.math.min
+
+internal class VitalMonitor {
+
+    private var lastKnownSample: Double = Double.NaN
+
+    private val listeners: MutableMap<VitalListener, VitalInfo> = mutableMapOf()
+
+    fun onNewSample(value: Double) {
+        lastKnownSample = value
+        notifyListeners(value)
+    }
+
+    fun getLastSample(): Double {
+        return lastKnownSample
+    }
+
+    fun register(listener: VitalListener) {
+        val value = lastKnownSample
+        synchronized(listeners) {
+            listeners[listener] = VitalInfo.EMPTY
```

Comment:
why do you want to reset the listeners `VitalInfo` here ?

Reply:
Because before this is called, we don't have any listener